arangodb-net-standard/Transport/Http/HttpClientConfig.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Linq;
+
+namespace ArangoDBNetStandard.Transport.Http
+{
+    public class HttpClientConfig
+    {
+        public Uri BaseAddress {
+            set {
+                _config += client => client.BaseAddress = value;
+                _key["baseaddress"] = value.ToString();
+            }
+        }
+        public MediaTypeHeaderValue ContentType { 
+            get {
+                return new MediaTypeHeaderValue(_contentTypeMap[_contentType]);
+            }
+        }
+
+        public void UseContentType(HttpContentType contentType)
+        {
+            _contentType = contentType;
+            _config += client =>
+            {
+                client.DefaultRequestHeaders.Accept.Clear();
+                client.DefaultRequestHeaders.Accept.Add(
+                    new MediaTypeWithQualityHeaderValue(_contentTypeMap[_contentType]));
+            };
+            _key["type"] = contentType.ToString();
+        }
+
+        private Dictionary<string, string> _key = new Dictionary<string, string>();
+        private Action<HttpClient> _config = _ => { };
+        private Action<HttpClient> _auth = _ => { };
+
+        private HttpClient Create() 
+        {
+            var cli = new HttpClient();
+            _config(cli);
+            _auth(cli);
+            return cli;
+        }
+
+        private HttpContentType _contentType;
+
+        private static readonly Dictionary<HttpContentType, string> _contentTypeMap =
+            new Dictionary<HttpContentType, string>
+            {
+                [HttpContentType.Json] = "application/json",
+                [HttpContentType.VPack] = "application/x-velocypack"
+            };
+
+        public void SetBasicAuth(string username, string password)
+        {
+            _auth = client =>
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(
+                    "Basic",
+                    Convert.ToBase64String(
+                        Encoding.ASCII.GetBytes($"{username}:{password}")));
+            _key["auth"] = $"basic,{username}:{password}";
+        }
+
+        public void SetJwtToken(string jwt)
+        {
+            _auth = client =>
+            {
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(
+                   "bearer",
+                   jwt);
+            };
+            _key["auth"] = $"jwt,{jwt}";
+        }
+
+        static Dictionary<string, HttpClient> _pool = new Dictionary<string, HttpClient>();
```

Comment:
I think we should move this to a dedicated `HttpClientPool` class that can control the number of open `HttpClient` instances. WDYT?

Reply:
Well here's the thing with the HttpClient... basically, it has two semantically different ways of being used. One is by means of async (as this program does) at which point the actual HttpClient is only a template that is used to generate the specific communications. For as far as the usage is kept within those (which are easy to identify because they are also the thread-safe parts), there is no concept of an HttpClient being "Open".

So I don't agree because since the usage is effectively constrained to the async versions, we would create more code thus a greater surface for errors and security issues to appear, plus we would be making the implicit statement that we are okay with the alternative semantics for HttpClient (and we are not).

Reply:
What I think I should do, is make the Build() method internal instead of public. This way, we make sure that there is no possibility of using the HttpClient from outside this driver, and thus no support for an "Open" state. Please share your comments.